api/test/test_tasks.py

```diff
@@ -0,0 +1,221 @@
+
+        response2 = client.post('/tasks', task_post_data, format='json')
+
+        self.assertEqual(response2.status_code, 409)
+        self.assertEqual(response2.data['detail'], 'Task `unique` field conflict')
+
+    def test_update_task(self):
+        task_post_data = {
+            'task_def': self.task_def_name,
+            'unique': 'classifer-2343',
+            'data': {
+                'foo': 'bar'
+            }
+        }
+
+        client = APIClient()
+        client.credentials(HTTP_AUTHORIZATION=self.token)
+
+        create_response = client.post('/tasks', task_post_data, format='json')
+
+        self.assertEqual(create_response.status_code, 201)
+
+        update = create_response.data
+        update['priority'] = 'high'
+
+        update_response = client.put('/tasks/' + str(update['id']), update, format='json')
+
+        self.assertEqual(update_response.status_code, 200)
+        self.assertEqual(list(update_response.data.keys()), task_keys)
+        self.assertEqual(update_response.data['priority'], 'high')
+
+    def test_update_task_auth(self):
+        task_post_data = {
+            'task_def': self.task_def_name,
+            'unique': 'classifer-2343',
+            'data': {
+                'foo': 'bar'
+            }
+        }
+
+        client = APIClient()
+        client.credentials(HTTP_AUTHORIZATION=self.token)
+
+        create_response = client.post('/tasks', task_post_data, format='json')
+
+        self.assertEqual(create_response.status_code, 201)
+
+        client = APIClient() # clear token
+
+        update = create_response.data
+        update['priority'] = 'high'
+
+        update_response = client.put('/tasks/' + str(update['id']), update, format='json')
+
+        self.assertEqual(update_response.status_code, 401)
+        self.assertEqual(update_response.data, {'detail': 'Authentication credentials were not provided.'})
+
+    def test_list_tasks(self):
+        task_post_data = {
+            'task_def': self.task_def_name,
+            'data': {
+                'foo': 'bar'
+            }
+        }
+
+        client = APIClient()
+        client.credentials(HTTP_AUTHORIZATION=self.token)
+
+        task_1_repsonse = client.post('/tasks', task_post_data, format='json')
+        task_2_response = client.post('/tasks', task_post_data, format='json')
+
+        client = APIClient() # clear token
+
+        list_response = client.get('/tasks')
+
+        self.assertEqual(list_response.status_code, 200)
+        self.assertEqual(list(list_response.data.keys()), ['count',
+                                                           'next',
+                                                           'previous',
+                                                           'results'])
+        self.assertEqual(len(list_response.data['results']), 2)
+        self.assertEqual(list(list_response.data['results'][0].keys()), task_keys)
```

Comment:
`.keys()` is not essential here, but okay for explicitness. In other words, taken a list of a dict, takes a list of the keys.